resource/menu/client/cl_main_page.lua

```diff
@@ -63,42 +63,91 @@ if isMenuDebug then
     end)
 end
 
+
+
 local oldVehVelocity = 0.0
+
+---if ped is in any vehicle then delete
+local GetVehicle = function()
+    local ped = PlayerPedId()
+    local oldVeh = GetVehiclePedIsIn(ped, false)
+    if oldVeh and oldVeh > 0 then
+        oldVehVelocity = GetEntityVelocity(oldVeh)
+        DeleteVehicle(oldVeh)
+    end
+
+end
+
 RegisterNUICallback('spawnVehicle', function(data, cb)
     if type(data) ~= 'table' then error("Invalid spawnVehicle NUI callback data") end
     local model = data.model
     if type(model) ~= 'string' then return end
-    if not IsModelValid(model) or not IsModelAVehicle(model) then
-        debugPrint("^1Invalid vehicle model requested: " .. model)
-        cb({ e = true })
-    else
-        local VehicleType = GetVehicleClassFromName(model)
-        local types = {
-            [8] = "bike",
-            [11] = "trailer",
-            [13] = "bike",
-            [14] = "boat",
-            [15] = "heli",
-            [16] = "plane",
-            [21] = "train",
-        }
-        local modelType = types[VehicleType] or "automobile"
-        if model == GetHashKey("submersible") or model == GetHashKey("submersible2") then
-            modelType = "submarine"
+
+    if not RedM then -- rdr3 support
+        if isModelValid(model) then
+
+            local VehicleType = GetVehicleClassFromName(model)
+            local types = {
+                [8] = "bike",
+                [11] = "trailer",
+                [13] = "bike",
+                [14] = "boat",
+                [15] = "heli",
+                [16] = "plane",
+                [21] = "train",
+            }
+            local modelType = types[VehicleType] or "automobile"
+            if model == GetHashKey("submersible") or model == GetHashKey("submersible2") then
+                modelType = "submarine"
+            end
+            -- collect the old velocity
+            GetVehicle()
+            TriggerServerEvent('txAdmin:menu:spawnVehicle', model, modelType)
+            cb({})
         end
-        -- collect the old velocity
-        local ped = PlayerPedId()
-        local oldVeh = GetVehiclePedIsIn(ped, false)
-        if oldVeh and oldVeh > 0 then
-            oldVehVelocity = GetEntityVelocity(oldVeh)
-            DeleteVehicle(oldVeh)
+    else
+        if not IsModelValid(model) then -- alow spawning horses or animals so we dont check if is a vehicle
+            debugPrint("^1Invalid  model requested: " .. model)
+            cb({ e = true })
         end
 
-        TriggerServerEvent('txAdmin:menu:spawnVehicle', model, modelType)
+        if IsModelAVehicle(model) then
+            ModelType = "vehicle"
+            GetVehicle()
+        else
+            ModelType = "ped" -- allow animals or peds
```

Comment:
Are any player "vehicles" considered ped/animals in RedM? If this is solely for arbitrary ped spawning, it may be better to just add a base menu option supporting that.

Reply:
@tabarra You see value in making this another option?

Reply:
vehicles can be  boats or wagons, peds horses animals, they all are created using the same native so i just named it ped
sure would be nice to have separated things in menu where users can know what can they spawn.